ptxcompiler/patch.py

```diff
@@ -183,7 +206,24 @@ def patch_needed():
     logger.debug("CUDA Driver version %s.%s" % driver_version)
     logger.debug("CUDA Runtime version %s.%s" % runtime_version)
 
-    return driver_version < runtime_version
+    return driver_version, runtime_version
+
+
+def safe_get_versions():
+    if check_disabled_in_env():
+        try:
+            # allow user to specify driver/runtime versions manually, if necessary
+            driver_version = os.environ["PTXCOMPILER_KNOWN_DRIVER_VERSION"].split(".")
+            runtime_version = os.environ["PTXCOMPILER_KNOWN_RUNTIME_VERSION"].split(".")
+        except KeyError:
+            warnings.warn(
+                "No way to determine driver and runtime versions for patching, "
+                "set PTXCOMPILER_KNOWN_DRIVER_VERSION/PTXCOMPILER_KNOWN_RUNTIME_VERSION"
+            )
+            return NO_DRIVER
```

Comment:
In ptxcompiler our logic is right now is

If `PTXCOMPILER_APPLY_NUMBA_CODEGEN_PATCH` is _not_ in the environment and `PTXCOMPILER_CHECK_NUMBA_CODEGEN_PATCH_NEEDED=0` then we don't call a subprocess and don't do any patching (and continue assuming that things are fine).

Here, if `PTXCOMPILER_CHECK_NUMBA_CODEGEN_PATCH_NEEDED=0` we need to communicate something back to `strings_udf`. So what is safe? We can return driver and runtime versions which will always result in no patching occuring (the sentinel values here). Or we can return extrinsically specified ones.

Perhaps it is safer to raise if we have been asked not to check but don't extrinsically specify the versions. I would also be happy with this (and can add appropriate docs to dask-cuda)

Reply:
Can we do something like, inside strings udf, `from ptxcompiler.patch import NO_DRIVER` and then check for that value exactly rather than relying on the sentinel values themselves triggering or not triggering the patch?